riscv/processor.cc

```diff
@@ -756,6 +757,7 @@ void processor_t::set_virt(bool virt)
      * The virtualized sstatus register also relies on this TLB flush,
      * since changing V might change sstatus.MXR and sstatus.SUM.
      */
+    state.prev_v = state.v;
```

Comment:
I think I've talked myself into accepting this approach.

Because etrigger only fires on an exception, and because an exception always calls `set_privilege()` and `set_virt()`, the previous privilege used for the etrigger check will always be the privilege of the immediately preceding instruction.

However I suspect there might still be a bug here. Scenario:
1. SRET from HS to VU: `prev_v` is set to 0
2. Trap from VU to VS: because `state.v` is not changing, `prev_v` doesn't change and remains 0
3. etrigger that's set to break on a VU-mode trap won't match properly because `prev_v` is incorrect

I think this can be solved simply by removing the conditional `if (state.v != virt)` around these two lines.

Reply:
Good catch; changed.

> Can you tell us how you've tested this? Do you have a failing test case that's now passing after this change? Does it include an etrigger breakpoint on a trap that goes from VU to VS mode?M

My testing didn't involve VU/VS mode (the testing was focused on U/S triggers for ECALLs / interrupts). Note that the original patch didn't have a prev_v, and it was added in response to feedback. However,  the scenario you described above seems perfectly viable.